src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs

```diff
@@ -191,15 +295,144 @@ public static async Task<WwwAuthenticateParameters> CreateFromResourceResponseAs
+            HttpResponseMessage httpResponseMessage = await httpClient.GetAsync(resourceUri, cancellationToken).ConfigureAwait(false);
+            var wwwAuthParams = CreateFromAuthenticationHeaders(httpResponseMessage.Headers);
+            return wwwAuthParams;
+        }
+
+        /// <summary>
+        /// Create WWW-Authenticate parameters from the HttpResponseHeaders for each auth scheme.
+        /// </summary>
+        /// <param name="httpResponseHeaders">HttpResponseHeaders.</param>
+        /// <returns>The parameters requested by the web API.</returns>
+        /// <remarks>Currently it only supports the Bearer scheme</remarks>
+        public static IReadOnlyList<WwwAuthenticateParameters> CreateFromAuthenticationHeaders(
+            HttpResponseHeaders httpResponseHeaders)
+        {
+            List<WwwAuthenticateParameters> parameterList = new List<WwwAuthenticateParameters>();
+
+            foreach (AuthenticationHeaderValue wwwAuthenticateHeaderValue in httpResponseHeaders.WwwAuthenticate)
+            {
+                try
+                {
+                    var parameters = CreateFromWwwAuthenticationHeaderValue(wwwAuthenticateHeaderValue.Parameter, wwwAuthenticateHeaderValue.Scheme);
+                    parameterList.Add(parameters);
+                }
+                catch (Exception ex) when (ex is not MsalException)
+                {
+                    throw new MsalClientException(MsalError.UnableToParseAuthenticationHeader, MsalErrorMessage.UnableToParseAuthenticationHeader, ex);
+                }
+            }
+
+            return parameterList;
+        }
+        #endregion Multi Scheme Api
+
+        /// <summary>
+        /// Creates parameters from the WWW-Authenticate string.
+        /// </summary>
+        /// <param name="wwwAuthenticateValue">String contained in a WWW-Authenticate header.</param>
+        /// <param name="scheme">Auth scheme of the result.</param>
+        /// <returns>The parameters requested by the web API.</returns>
+        private static WwwAuthenticateParameters CreateFromWwwAuthenticationHeaderValue(string wwwAuthenticateValue, string scheme)
+        {
+            if (string.IsNullOrWhiteSpace(wwwAuthenticateValue))
+            {
+                throw new ArgumentNullException(nameof(wwwAuthenticateValue));
+            }
+
+            //Special NTLM case does not have an a=b format
+            if (scheme == "NTLM")
+            {
+                return new WwwAuthenticateParameters
+                {
+                    RawParameters = new Dictionary<string, string>()
+                    {
+                        { scheme, wwwAuthenticateValue }
+                    },
+                    AuthScheme = scheme
+                };
+            }
+
+            IDictionary<string, string> parameters;
+
+            var authValuesSplit = wwwAuthenticateValue.Split(new char[] { ' ' }, 2);
```

Comment:
I think there is an overload of Split that takes into account all whitespace chars.